### PR TITLE
Boat yard

### PR DESCRIPTION
**Pull request by Sunaina04**

### CODE REVIEW

File: src/Type/CommonType.ts [hunk 1]
```diff
@@ -43,5 +43,20 @@ export interface FormFieldsProps {
...
 export type NullableDateArray = (Date | null)[] | null
+
+export interface Role {
+  name: string
+}
+
+export interface Country {
+  id: number
+  name: string
+  label: undefined
```
Comment (ShKushal) [inline src/Type/CommonType.ts:55]: will be only undefined ?
then why you need it ?

File: src/Type/ApiTypes.ts [hunk 2]
```diff
@@ -309,3 +321,21 @@ export interface WorkOrderResponse {
     WorkOrderPayload: WorkOrderPayload
   }
 }
+
+export interface ContentData {
+  name: string
+  id: number
+  label: undefined
```
Comment (ShKushal) [inline src/Type/ApiTypes.ts:328]: same here , why undefined only ?

File: src/Components/Moormanage/Boatyards/AddBoatyards.tsx [hunk 3]
```diff
@@ -82,55 +163,76 @@ const AddBoatyards = () => {
...
+
+          <div>
+            <div>
+              <div className="mt-2">
+                <InputComponent
+                  value={zipCode}
+                  onChange={(e) => setZipCode(e.target.value)}
+                  placeholder="Zip code"
+                  style={{
+                    width: '13vw',
```
Comment (ShKushal) [inline src/Components/Moormanage/Boatyards/AddBoatyards.tsx:192]: remove repetition of code by create compoenent

File: src/Components/Moormanage/Boatyards/AddBoatyards.tsx [hunk 4]
```diff
@@ -4,8 +4,22 @@ import { DataTable } from 'primereact/datatable'
...
+import { Dropdown } from 'primereact/dropdown'
 
 const AddBoatyards = () => {
+  const [boatyardId, setBoatyardId] = useState('')
+  const [boatyardName, setBoatyardName] = useState('')
+  const [emailAddress, setEmailAddress] = useState('')
+  const [phone, setPhone] = useState('')
+  const [address, setAddress] = useState('')
+  const [aptSuite, setAptSuite] = useState('')
+  const [state, setState] = useState(null)
```
Comment (ShKushal) [inline src/Components/Moormanage/Boatyards/AddBoatyards.tsx:16]: remove null

File: src/Components/Moormanage/Boatyards/AddBoatyards.tsx [hunk 5]
```diff
@@ -4,8 +4,22 @@ import { DataTable } from 'primereact/datatable'
...
+  const [country, setCountry] = useState(null)
+  const [zipCode, setZipCode] = useState('')
+  const [mainContact, setMainContact] = useState('')
+
+  const handleSave = () => {}
```
Comment (ShKushal) [inline src/Components/Moormanage/Boatyards/AddBoatyards.tsx:21]: return void instead

**Review by ShKushal — CHANGES_REQUESTED** [review]

File: src/Type/CommonType.ts (hunk 1 above)
Comment (Sunaina04) [reply src/Type/CommonType.ts:55]: actually getting null in response , that's why making it undefined for now

File: src/Components/Moormanage/Boatyards/AddBoatyards.tsx (hunk 3 above)
Comment (Sunaina04) [reply src/Components/Moormanage/Boatyards/AddBoatyards.tsx:192]: already using CustomInputComponent here , code for styling is repeating here , i tried like this now -> 
  const style = {
    width: '13vw',
    height: '4vh',
    border: '1px solid gray',
    borderRadius: '0.50rem',
    fontSize: '0.80vw',
  }

File: src/Type/ApiTypes.ts (hunk 2 above)
Comment (Sunaina04) [reply src/Type/ApiTypes.ts:328]: i don't know the type actually, getting null in response ,changing into string for now, will change later once i got value in response

**Review by ShKushal — APPROVED** [review]

**Pull request merged**